CRM/Report/Form.php

```diff
@@ -2637,23 +2643,22 @@ public function buildQuery($applyLimit = TRUE) {
    * Build group by clause.
    */
   public function groupBy() {
-    $groupBys = array();
     if (!empty($this->_params['group_bys']) &&
       is_array($this->_params['group_bys'])
     ) {
       foreach ($this->_columns as $tableName => $table) {
         if (array_key_exists('group_bys', $table)) {
           foreach ($table['group_bys'] as $fieldName => $field) {
             if (!empty($this->_params['group_bys'][$fieldName])) {
-              $groupBys[] = $field['dbAlias'];
+              $this->_groupByArray[] = $field['dbAlias'];
```

Comment:
@totten the php7.1 error is that we are trying to use [] to add variables into an array of groupBy or OrderBys see https://github.com/civicrm/civicrm-core/pull/10765/files#diff-fcb0ad5f41ccfc50380e03cda321d528R382 however the variable has been initalise  as a string not arrays in the cases Eileen is highlighting in this PR

This PR goes some way as to ensuring that $this->_groupBy should always be the string version and $this->_groupByArray is the array to build $this->_groupBy